src/test/java/org/assertj/core/api/assumptions/Assumptions_assumeThat_Atomics_Test.java

```diff
@@ -17,178 +17,167 @@
+                         assumeThat(actual).hasValue("test");
+                       }
+                     },
+                     new AssumptionRunner<AtomicReferenceArray<String>>(new AtomicReferenceArray<>(array("2", "5", "7"))) {
+                       @Override
+                       public void runFailingAssumption() {
+                         assumeThat(actual).contains("20");
+                       }
+
+                       @Override
+                       public void runPassingAssumption() {
+                         assumeThat(actual).contains("7");
+                       }
+                     },
+                     new AssumptionRunner<AtomicReferenceFieldUpdater<VolatileFieldsHolder, String>>(AtomicReferenceFieldUpdater.newUpdater(VolatileFieldsHolder.class,
+                                                                                                                                            String.class,
+                                                                                                                                            "stringValue")) {
+                       @Override
+                       public void runFailingAssumption() {
+                         assumeThat(actual).hasValue("other", VOLATILE_FIELDS_HOLDER);
+                       }
+
+                       @Override
+                       public void runPassingAssumption() {
+                         assumeThat(actual).hasValue("test", VOLATILE_FIELDS_HOLDER);
+                       }
+                     },
+                     new AssumptionRunner<AtomicMarkableReference<String>>(new AtomicMarkableReference<>("test", true)) {
+                       @Override
+                       public void runFailingAssumption() {
+                         assumeThat(actual).hasReference("other");
+                       }
+
+                       @Override
+                       public void runPassingAssumption() {
+                         assumeThat(actual).hasReference("test");
+                       }
+                     },
+                     new AssumptionRunner<AtomicStampedReference<String>>(new AtomicStampedReference<>("test", 1)) {
+                       @Override
+                       public void runFailingAssumption() {
+                         assumeThat(actual).hasStamp(0);
+                       }
+
+                       @Override
+                       public void runPassingAssumption() {
+                         assumeThat(actual).hasStamp(1);
+                       }
+                     });
   }
 
   @ParameterizedTest
   @MethodSource("provideAssumptionsRunners")
   void should_ignore_test_when_assumption_fails(AssumptionRunner<?> assumptionRunner) {
-    assertThatExceptionOfType(AssumptionViolatedException.class).isThrownBy(() -> assumptionRunner.runFailingAssumption());
+    assertThatExceptionOfType(TestAbortedException.class).isThrownBy(assumptionRunner::runFailingAssumption);
```

Comment:
let's replace all `assertThatExceptionOfType(TestAbortedException.class).isThrownBy` by the `expectAssumptionNotMetException` construct.